File: reverseArray.js
const reverseArray1 = (arr) => {
    let temp = 0
    for (left = 0, right = (arr.length - 1); left < right; left++, right--) {
        temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
    }
    return arr;
}
q = [1, 2, 3, 4, 5]
console.log(reverseArray1(q));
console.log('__________10/23/17__________');

const reverseArray2 = (arr) => {
    let temp = 0
    for (i = 0; i < arr.length / 2; i++) {
        temp = arr[i];
        arr[i] = arr[(arr.length - 1) - i];
        arr[(arr.length - 1) - i] = temp;
    }
    return arr;
}
console.log(reverseArray2(q));
console.log('__________10/23/17__________');

///////////////////////////////////////////

let w = [9, 8, 7, 6]

const reverseArray3 = (arr) => {
    let temp = 0;
    for (left = 0, right = arr.length - 1; left < right; left++, right--) {
        temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
    }
    return arr;
}
console.log(reverseArray3(w));
console.log('__________10/26/17__________');

const reverseArray4 = (arr) => {
    let temp = 0;
    for (i = 0; i < arr.length / 2; i++) {
        temp = arr[i];
        arr[i] = arr[(arr.length - 1) - i];
        arr[(arr.length - 1) - i] = temp
    }
    return arr;
}
console.log(reverseArray4(w));
console.log('__________10/26/17__________');

///////////////////////////////////////////

let e = [1, 3, 5, 7, 9]

const reverseArray5 = (arr) => {
    let temp = 0;
    for (left = 0, right = arr.length - 1; left < right; left++, right--) {
        temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
    }
    return arr;
}
console.log(reverseArray5(e));
console.log('__________10/31/17__________');

const reverseArray6 = (arr) => {
    let temp = 0;
    for (i = 0; i < arr.length / 2; i++) {
        temp = arr[i];
        arr[i] = arr[(arr.length - 1) - i];
        arr[(arr.length - 1) - i] = temp;
    }
    return arr;
}
console.log(reverseArray6(e));
console.log('__________10/31/17__________');

///////////////////////////////////////////

File: reverseArray.py
def reverseArray1(list):
    for i in range (0, len(list) / 2):
        temp = list[i]
        list[i] = list[(len(list) - 1) - i]
        list[(len(list) - 1) - i] = temp
    return list
q = [1, 2, 3, 4, 5]
print reverseArray1(q)
print '__________10/23/17__________'

def reverseArray2(list):
    left = 0
    right = len(list) - 1
    while left < right:
        temp = list[left]
        list[left] = list[right]
        list[right] = temp
        left += 1
        right -= 1
    return list
print reverseArray2(q)
print '__________10/23/17__________'

####################################

w = ['a', 'b', 'c', 'd']

def reverseArray3(list):
    for i in range (0, len(list) / 2):
        temp = list[i]
        list[i] = list[(len(list) - 1) - i]
        list[(len(list) - 1) - i] = temp
    return list
print reverseArray3(w)
print '__________10/26/17__________'

def reverseArray4(list):
    left = 0
    right = len(list) - 1
    while left < right:
        temp = list[left]
        list[left] = list[right]
        list[right] = temp
        left += 1
        right -= 1
    return list
print reverseArray4(w)
print '__________10/26/17__________'

####################################

e = ['tom', 'dick', 'harry']

def reverseArray5(list):
    for i in range (0, len(list) / 2):
        temp = list[i]
        list[i] = list[(len(list) - 1) - i]
        list[(len(list) - 1) - i] = temp
    return list
print reverseArray5(e)
print '__________10/31/17__________'

def reverseArray6(list):
    left = 0
    right = len(list) - 1
    while left < right:
        temp = list[left]
        list[left] = list[right]
        list[right] = temp
        left += 1
        right -= 1
    return list
print reverseArray6(e)
print '__________10/31/17__________'
